src/common/actions/snap-builds.js

```diff
@@ -18,27 +20,38 @@ const REQUEST_POST_OPTIONS = {
   credentials: 'same-origin'
 };
 
-export function fetchBuildsSuccess(builds) {
+export function fetchBuildsSuccess(id, builds) {
   return {
     type: FETCH_BUILDS_SUCCESS,
-    payload: builds
+    payload: {
+      id,
+      builds
+    }
   };
 }
 
-export function fetchBuildsError(error) {
+export function fetchBuildsError(id, error) {
```

Comment:
Is there a need to store information about fetching a specific build?

We are fetching build data separately because of an implementation detail: we only have an API endpoint for fetching data on builds singularly, not in batch. This means that anything can go wrong with any of the API requests, but is there a feature in our application that relies on the success or failure of individual requests?